predict.py

```diff
@@ -166,8 +206,39 @@ def predict(
             n_tokens += 1
             yield decoded_token
         t = time.time() - st
-
+        print(f"hostname: {socket.gethostname()}")
         if debug:
             print(f"cur memory: {torch.cuda.memory_allocated()}")
             print(f"max allocated: {torch.cuda.max_memory_allocated()}")
             print(f"peak memory: {torch.cuda.max_memory_reserved()}")
+
+    # def remove(f: "Callable", defaults: "dict[str, Any]") -> "Callable":
+    #     # pylint: disable=no-self-argument
+    #     # for the purposes of inspect.signature as used by predictor.get_input_type,
+    #     # remove the argument (system_prompt)
+    #     wrapped = functools.partialmethod(f, **defaults)
+    #     sig = inspect.signature(wrapped)
+    #     # TypeError: functools.partialmethod(<function Predictor.predict at 0x7fa5d2136340>, , system_prompt=None) is not a callabl object
+    #
+    #     params = [p for name, p in sig.parameters.items() if name not in defaults]
+    #     wrapped.__signature__ = sig.replace(parameters=params)
+    #     return wrapped
+
+    # if not USE_SYSTEM_PROMPT:
+    #     predict = remove(predict, {"system_prompt": None})
+
+    _predict = predict
+
+    def base_predict(self, *args, **kwargs) -> ConcatenateIterator:
+        kwargs["system_prompt"] = None
+        return self._predict(*args, **kwargs)
+
+    # for the purposes of inspect.signature as used by predictor.get_input_type,
+    # remove the argument (system_prompt)
```

Comment:
This is good - I'd maybe add a higher level comment for any non-power users who stumble into this codebase, something to the effect of `This removes `system_prompt` from the Replicate API for non-chat models`.

Interesting to think about how to only surface a subset of `predict` arguments in the API - I think the subclassing I was talking about earlier works if we want to keep around multiple inference engines, but it's repetitive for defining arguments. 

I suppose if we have multiple configurations of the same `predict` we can just make the params to scrub from the signature part of the `config.py` for each model instead of keying off of `USE_SYSTEM_PROMPT`

Reply:
ah,`remove()` generalizes this! that's what I get for not reading commented methods 🙃

Reply:
yes, exactly!